curriculum/challenges/english/25-front-end-development/workshop-fruit-search-app/67ed044dc76cf4a441618f2c.md

```diff
@@ -0,0 +1,123 @@
+---
+id: 67ed044dc76cf4a441618f2c
+title: Step 5
+challengeType: 0
+dashedName: step-5
+---
+
+# --description--
+
+Create a `query` state variable and its setter function, `setQuery`, using the `useState` Hook.
+
+# --hints--
+
+You should use the array destructuring syntax for the `query` and `setQuery` variables.
+
+```js
+assert.match(code, /(const|let)\s+\[\s*(query)\s*,\s*(setQuery)\s*\]\s*/);
+```
+
+You should use the `useState` Hook.
+
+```js
+assert.match(code, /useState\(/);
```

Comment:
it looks like the other labs and workshops are using this approach for testing.
Make sure this works on your end but this is the test used from the color picker lab

```js
async () => {
    const abuseState = __helpers.spyOn(React, "useState");
    const script = [...document.querySelectorAll("script")].find((s) => s.dataset.src === "index.jsx").innerText;
    const exports = {};
    const _a = eval(script);
    const _b = await __helpers.prepTestComponent(exports.ColorPicker);
    assert.isAtLeast(abuseState.calls.length, 1);
}
```